job/job.go

```diff
@@ -0,0 +1,230 @@
+package job
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"os/exec"
+	"sync"
+	"syscall"
+
+	"github.com/gopheryan/jobby/streamer"
+)
+
+// Current process state
+type State string
+
+const (
+	// The process is currently running
+	JobStatusRunning State = "RUNNING"
+	// Not necessarily a 'success' (see exit code for that)
+	// This state signals that the process is no longer running
+	JobstatusComplete State = "COMPLETE"
+	// Different from 'COMPLETE' in that this state
+	// means that the user deliberately stopped the job
+	JobStatusStopped State = "STOPPED"
+)
+
+func NewState(processExited bool, e *exec.ExitError) State {
+	if processExited {
+		// ExitCode() returns -1 if the process hasn't exited or was terminated by a signal
+		// Well we know the process exited
+		if e != nil && e.ExitCode() == -1 {
+			status, ok := e.ProcessState.Sys().(syscall.WaitStatus)
+			if ok && status.Signal() == syscall.SIGKILL {
+				// I suppose we can't know for certain that it this program who sent
+				// the SIGKILL, but it was *probably* us
+				// Maybe the documentation around the "STOPPED" state should instead be
+				// that this state means the program was forcefully terminated, rather than
+				// terminated *by a user*
+				return JobStatusStopped
+			}
+		}
+		return JobstatusComplete
+	}
+	return JobStatusRunning
+}
+
+type Status struct {
+	CurrentState State
+	ReturnCode   *int
+}
+
+type JobArgs struct {
+	Command    string
+	Args       []string
+	StdoutPath string
+	StderrPath string
+}
+
+type Job struct {
+	sync.Mutex
+	cmd           exec.Cmd
+	processExited bool
+	exitErr       *exec.ExitError
+
+	stdoutPath string
+	stderrPath string
+}
+
+func logFileClose(f *os.File) {
+	if err := f.Close(); err != nil {
+		if !errors.Is(err, os.ErrInvalid) {
+			slog.Error(fmt.Sprintf("Failed to close file: %s", err))
```

Comment:
Suggestion: embrace structured logging 
```suggestion
			slog.Error("Failed to close file", "error", err)
```